### PR TITLE
Redesign logs tab

### LINKED ISSUE #1176: Redesign logs tab of Extension's Option page
The design of logs tab needs a redesign bump as it was not build based on a specific design, rather some minor discussions

### PR DESCRIPTION
**Pull request by wirednkod**
Following figma designs this PR redesigns the logs tab.

Fixes #1176

### CODE REVIEW

**Comment by tomaka** [comment]
The link shows 5 different designs, which one is it?

**Comment by wirednkod** [comment]
> The link shows 5 different designs, which one is it?

I thought the link would take one to the correct one. Its the last of the list at the bottom

**Comment by tomaka** [comment]
Can I ask for some adjustments?

- Use a monospace fronts. Logs use monospace fonts. It's almost a rule of life.
- Don't use bold/italic/whatever, otherwise you're disturbing the monospace font. If you want to highlight warnings/errors, it should be with colors for example.
- The log levels (debug, warning, ...) should all have the same number of characters. Again, to not disturb the alignment. Use "trace", "debug", "info " (with a space), "warn " (with a space), "error"

**Comment by wirednkod** [comment]
> * Use a monospace fronts. Logs use monospace fonts. It's almost a rule of life.
> * Don't use bold/italic/whatever, otherwise you're disturbing the monospace font. If you want to highlight warnings/errors, it should be with colors for example.

For the above cc @Goranch3 

Will add this:
> The log levels (debug, warning, ...) should all have the same number of characters. Again, to not disturb the alignment. Use "trace", "debug", "info " (with a space), "warn " (with a space), "error"

**Comment by Goranch3** [comment]
Use a monospace fronts. Logs use monospace fonts. It's almost a rule of life. 
**OK**
Don't use bold/italic/whatever.... **ok**
it should be with colors for example. **color alone is not enough of an indicator, especially if the information is important**

**Comment by Goranch3** [comment]
@tomaka @wirednkod can you please give me a realistic piece of a log, with exact length, thanks

**Comment by tomaka** [comment]
> especially if the information is important

The information is not important.

We're talking about logs. I don't disagree that conceptually color isn't enough of an indicator, but you can't just apply this blindly to any context. We're talking about logs here. The point is not to report an information. The information in question is already indicated through the words "debug", "warn", etc. The point is to highlight specific lines so that they can easily be spotted while scrolling through logs.

The mere fact that we have a logs tag is a UX horror. Users aren't supposed to see this tab.

> [wirednkod](https://github.com/wirednkod) requested a review from [tomaka](https://github.com/tomaka) 20 hours ago

Did you make changes? Is there some kind of screenshot or something that would allow me to do the review, like you're asking me to do? I don't know if you can but I'm not capable of reading HTML and CSS class names and guessing how it looks like.

**Comment by wirednkod** [comment]
> Did you make changes? Is there some kind of screenshot or something that would allow me to do the review, like you're asking me to do? I don't know if you can but I'm not capable of reading HTML and CSS class names and guessing how it looks like.

There are changes based on the comments given, above:
> - Use a monospace fronts. Logs use monospace fonts. It's almost a rule of life.
> - Don't use bold/italic/whatever, otherwise you're disturbing the monospace font. 

Screenshot can be found below, though besides the comments' changes, I am following the figma proposal:

image

**Comment by wirednkod** [comment]
> We're talking about logs. I don't disagree that conceptually color isn't enough of an indicator, but you can't just apply this blindly to any context. We're talking about logs here. The point is not to report an information. The information in question is already indicated through the words "debug", "warn", etc. The point is to highlight specific lines so that they can easily be spotted while scrolling through logs.

@Goranch3 any input on that?

**Comment by Goranch3** [comment]
@tomaka can you give us a few screenshots of well designed logs so we can apply it to our case without wasting too much time?
Also
"The mere fact that we have a logs tab is a UX horror. Users aren't supposed to see this tab."
I really believe we don't need to communicate like this, this is not the first time you are seeing this, verbalize it timely let's discuss, where do you think logs should be, should we rethink the entire content of the options page?

**Comment by tomaka** [comment]
Here's what Grafana does: https://grafana.com/docs/grafana/latest/explore/logs-integration/#logs-navigation
It's very good but it has a ton of features that we don't have and don't need, such as searching and filtering.

Here's what VSCode shows in "Log" language mode, which is also decent:

image

> I really believe we don't need to communicate like this, this is not the first time you are seeing this, verbalize it timely let's discuss

The reason why I'm saying that the logs are a UX horror is that the content of the logs is incomprehensible for anyone who isn't a developer anyway. And thus it really doesn't matter if the average Joe doesn't understand for example what the buttons are for. In other words, it's out of scope for the logs tab to have good UX.

**Comment by Goranch3** [comment]
@tomaka ok this helps, how about the place where logs live, if they are not in their own tab where can we show them, keep in mind they require a lot of space, so it needs to be a separate page.

**Comment by tomaka** [comment]
The reason why we initially added logs was generally:

- Figure out why smoldot has crashed when it crashes.
- Add a way for chain developers to understand why things aren't working. Not necessarily understand by themselves, but copy logs to us so that we can read them.
- Understand bugs in substrate-connect.

However the first two points are now covered by https://github.com/paritytech/substrate-connect/pull/1228 and https://github.com/paritytech/substrate-connect/pull/1044, and the last point will hopefully become obsolete in the future
Consequently I'm not sure how useful a logs tab is anymore.

I'm not saying it can not be useful, but it probably is less than before.

I know we talked about some kind of "developer mode" before that could be activated, and I think it's the best way to do it. The "developer mode" could even have a secret way of being activated (such as manually tweaking the storage of the extension), not necessarily a UI button.

**Comment by Goranch3** [comment]
@tomaka OK is it sane to say that we will remove logs tab from the current version, and make a separate issue for Developers mode, so we don't need to worry were they live now?

**Comment by tomaka** [comment]
We've already released versions including this logs tab.
My opinion is to merge this PR, then add a "developer mode" and put the logs tab behind it.

**Comment by wirednkod** [comment]
> We've already released versions including this logs tab.
> My opinion is to merge this PR, then add a "developer mode" and put the logs tab behind it.

I agree. Once approved I ll merge this and create another PR with the developer mode button

File: projects/extension/src/background/index.ts [hunk 1]
```diff
@@ -140,6 +140,7 @@ const logger = (level: number, target: string, message: string) => {
 
   if (all.length >= 1000) all.shift()
   all.push(incLog)
+  return { all, warn, error }
```
[the remark this answers is not in the document]
Comment (wirednkod) [reply projects/extension/src/background/index.ts:143]: Thank you.. That was not supposed to be there

**Review by tomaka — APPROVED** [review]

**Pull request merged**